docs/CONTENT_ACCESS_SUMMARY.md

```diff
@@ -0,0 +1,280 @@
+- `admin_grant` - Admin gave free access
+
+**Key Columns:**
+- `user_id` - Who has access
+- `group_id` - Which group
+- `product_id` - Product id denotes the entity in Stripe that tracks an offering/product
+- `track_id` - Optional: Grants access to a specific track within a group
+- `role_id` - Optional: References `groups_roles` table, represents role-based access grants by admins
+- `access_type` - How they got access (stripe_purchase, admin_grant)
+- `stripe_session_id` - **Nullable** (only for Stripe purchases)
+- `amount_paid` - Amount paid (0 for free grants)
+- `status` - active/expired/revoked
+- `granted_by_id` - Admin who granted access (for admin grants)
+- `expires_at` - Optional expiration date
+- `metadata` - Flexible JSONB for additional info
+
+**Access Granularity:**
+Access can be granted at multiple levels:
+- **Group-level**: Just `group_id` set - access to entire group content
+- **Track-level**: `track_id` set - access to specific track content
+- **Role-level**: `role_id` set - access tied to a specific group role
+
+**Automatic Expiration Mirroring:**
+The `expires_at` value from `content_access` is automatically mirrored to related tables using PostgreSQL triggers:
+- **When track_id is NULL** (group-level or role-based): `group_memberships.expires_at` (based on `user_id` + `group_id`)
+- **When track_id is set**: `tracks_users.expires_at` (based on `user_id` + `track_id`)
+- **When role_id is set**: `group_memberships_group_roles.expires_at` (based on `user_id` + `group_id` + `group_role_id`)
+
+**Important:** Track purchases do NOT update `group_memberships.expires_at`. This prevents a one-off track purchase from overwriting a long-term group membership expiration. However, role bumps DO update group membership (having a role implies group access).
+
+This avoids the need for JOINs when checking expiration - you can query the respective tables directly.
+
+## Database Triggers for Automatic Expiration Sync
+
+### Overview
+To avoid constantly joining `content_access` with membership tables, PostgreSQL triggers automatically keep `expires_at` values in sync.
+
+### How It Works
+
+**When content access is granted or updated:**
+1. User inserts/updates a record in `content_access` with `expires_at` set
+2. Trigger `content_access_expires_at_sync` fires automatically
+3. Function `sync_content_access_expires_at()` executes:
+   - **If track_id is NULL**: Updates `group_memberships.expires_at` (group-level or role-based purchase)
+   - **If track_id is set**: Updates `tracks_users.expires_at` (track-specific purchase, does NOT update group_memberships)
+   - **If role_id is set**: Updates `group_memberships_group_roles.expires_at` (role-specific purchase, also updates group_memberships if no track_id)
+
+**When access is revoked or expires:**
+1. Status changes to 'revoked' or 'expired' in `content_access`
+2. Trigger `content_access_expires_at_clear` fires
+3. Function `clear_content_access_expires_at()` executes:
```

Comment:
so it always has an expires_at value if content is accessible? and for lifetime we just put one far in the future?